cookbooks/retention.md

```diff
@@ -0,0 +1,211 @@
+    ms
+    .groupby("period", "os")
+    .agg(F.countDistinct("client_id").alias("n_week_clients"))
+)
+
+retention_by_os = (
+    weekly_counts
+    .join(os_counts, on='os')
+    .withColumn("retention", F.col("n_clients") / F.col("total_count"))
+)
+```
+
+Peaking at 6-Week Retention
+
+```python
+retention_by_os.filter("period = 6").show()
+```
+
+```
++----------+------+--------------+-------------+------------------+
+|        os|period|n_week_clients|total_clients|         retention|
++----------+------+--------------+-------------+------------------+
+|     Linux|     6|          1495|         4196|0.3562917063870353|
+|    Darwin|     6|          1288|         2497|0.5158189827793352|
+|Windows_NT|     6|         29024|        60564|0.4792285846377386|
++----------+------+--------------+-------------+------------------+
+```
+
+we observe that 35.6% of Linux users whose profile was created in the first half of April submitted a ping 6 weeks later, and so forth. The example code snippets are consolidated in [this notebook](https://gist.github.com/benmiroglio/fc708e5905fad33b43adb9c90e38ebf4).
+
+
+# New vs. Existing User Retention
+
+The above example calculates **New User Retention**, which is distinct from **Existing User Retention**. This distinction is important when understanding retention baselines (i.e. does this number make sense?). Existing users typically have much higher retention numbers than new users. 
+
+Note that is more common in industry to refer to Existing User Retention as "Churn" (Churn = 1 - Retention), however, we use retention across the board for the sake of consistency and interpretability.
+
+**Please be sure to specify whether or not your retention analysis is for new or existing users.**
+
+
+
+
+
+
+
+
+
+# Confounding Factors
+
+When performing retention analysis between two or more groups, it is important to look at other usage metrics to get an understanding of other influential factors.
+
+For example, imagine you are tasked with reporting retention numbers for users that enabled sync (`sync_configured = true`) compared to users that haven't. You find that users with and without sync have a 1 week retention of 0.80 and 0.40, respectively. Wow--we should really be be promoting sync as it could double retention numbers!
+
+*Not quite*. Turns out you next look at `active_ticks` and `total_uri_count` and find that sync users report much higher numbers for these measures as well. Now how can we explain this difference in retention?
+
+There could be an entirely separate cookbook devoted to answering this question, however this contrived example is meant to demonstrate that simply comparing retention numbers between two groups isn't capturing the full story. Sans an experiment or model-based approach, all we can say is "enabling sync is **associated** with higher retention numbers." There is still value in this assertion, however it should be stressed that association/correlation != causation!
```

Comment:
Great explaination and example. It's worth bolding **correlation != causation** so the reader can get the gist of this example while scanning.

Reply:
good point!